pkg/controller/operators/olm/operator.go

```diff
@@ -252,6 +260,26 @@ func newOperatorWithConfig(ctx context.Context, config *operatorConfig) (*Operat
 			return nil, err
 		}
 
+		// Register QueueInformer for olmConfig
+		olmConfigInformer := externalversions.NewSharedInformerFactoryWithOptions(
+			op.client,
+			config.resyncPeriod(),
+		).Operators().V1().OLMConfigs().Informer()
+		olmConfigQueueInformer, err := queueinformer.NewQueueInformer(
+			ctx,
+			queueinformer.WithInformer(olmConfigInformer),
+			queueinformer.WithLogger(op.logger),
+			queueinformer.WithQueue(op.olmConfigQueue),
+			queueinformer.WithIndexer(olmConfigInformer.GetIndexer()),
+			queueinformer.WithSyncer(queueinformer.LegacySyncHandler(op.syncOLMConfig).ToSyncer()),
+		)
+		if err != nil {
+			return nil, err
+		}
+		if err := op.RegisterQueueInformer(olmConfigQueueInformer); err != nil {
+			return nil, err
+		}
+
```

Comment:
The OLMConfig controller needs to requeue csvs, which is done by adding an object to this queue. We could build and pass this queue into OLM if we wanted the olmConfig controller to be built with controller-runtime. Thoughts?

Reply:
One benefit to the existing implementation is that the [logic](https://github.com/operator-framework/operator-lifecycle-manager/pull/2466/files#diff-1c45679686ec7da9a4e3c02d009d84f465130fec2278b9512d27d53e23224f3aR1364) responsible for deleting the CSVs is in the same file.

One negative to the existing implementation is that the olmConfig controller may be extended to support other features which might be easier to implement / understand if written with controller runtime.

Reply:
I perused  moving this reconcile logic into controller-runtime, but all of OLM's controller-runtime operators are built with [a client that ignores copied csvs](https://github.com/operator-framework/operator-lifecycle-manager/blob/82fbb1aa0c391a2197e9a665759c3f94764850ab/cmd/olm/manager.go#L58-L60). In my opinion, the performance benefit of not having these controllers caching copied csvs outweighs those of moving towards controller-runtime. As an added benefit, we avoid introducing new controllers that really on CSVs and keep said logic within the olm controller.